src/walker/walker.ts

```diff
@@ -256,6 +256,7 @@ export class Walker extends EventEmitter<WalkerEmitter> {
   protected processFragment(): SchemaNode {
     const { walkingOptions, path } = this;
     let { fragment } = this;
```

Comment:
we'd take this fragment as an "original fragment"

```
const { fragment: initialFragment, walkingOptions, path } = this;  // or original, whatever, I suck at naming things.
let { fragment } = this;
```

Reply:
Good idea, I guess that would make it more generic